### DISCUSSION TITLE
Inpainting and refiner swapped

### DISCUSSION THREAD

**Discussion by emmagauthier01** [lllyasviel/Fooocus#1985] [General]
hello and bravo for this amazing software. is this possible to disable the Refiner Swapper during inpainting. It appears in step 15/30 and I would like to do without it on some project. Is this possible even by tampering with the code? I did not find while searching

**Comment by mashb1t** [comment]
if you don't define a refiner there will not be a refiner swap. Just select "None" in the refiner model selection.

**Comment by SJM-AI** [comment]
Try going into debug>Inpaint> turn the inpaint engine to none.

**Comment by wanglaofei** [comment]
You can modify the code in modules->async_worker.py do disable the refiner process.
![image](https://github.com/lllyasviel/Fooocus/assets/26437644/8b067f96-1b68-4620-bd4d-608ac0f7ff93)
set the use_synthetic_refiner = False

**Reply by RovingDog** [reply]
Thanks friend

**Reply by achmednasir** [reply]
thanks alot. this fixes most of the issues with inpainting and pony models, where the result had oversaturated colors and pixels
